libbeat/dashboards/decode.go

```diff
@@ -36,21 +38,26 @@ var (
 )
 
 // DecodeExported decodes an exported dashboard
-func DecodeExported(result common.MapStr) common.MapStr {
+func DecodeExported(exported []byte) []byte {
 	// remove unsupported chars
-	objects := result["objects"].([]interface{})
-	for _, obj := range objects {
-		o := obj.(common.MapStr)
+	result := make([]byte, 0)
+	scanner := bufio.NewScanner(bytes.NewReader(exported))
+	for scanner.Scan() {
+		o := common.MapStr{}
+		err := json.Unmarshal(scanner.Bytes(), &o)
+		if err != nil {
+			continue
+		}
 		for _, key := range responseToDecode {
 			// All fields are optional, so errors are not caught
 			err := decodeValue(o, key)
 			if err != nil {
 				logger := logp.NewLogger("dashboards")
 				logger.Debugf("Error while decoding dashboard objects: %+v", err)
 			}
+			result = append(result, []byte(o.String())...)
 		}
 	}
```

Comment:
This should check if `scanner.Err()` is non-nil. In this case I think it can only happen if it exceeds the MaxScanTokenSize.

Reply:
I want to get rid of the scanner in favour of the reader. This is just a leftover. Thanks for noticing it.